MeshLib/MeshSurfaceExtraction.cpp

```diff
@@ -378,21 +374,52 @@ bool MeshSurfaceExtraction::createSfcMeshProperties(
         return false;
     }
 
-    std::string const vec_name ("MaterialIDs");
-    if (!properties.existsPropertyVector<int>(vec_name, MeshLib::MeshItemType::Cell, 1))
+    std::array<MeshLib::MeshItemType, 2> const type{
+        {MeshLib::MeshItemType::Node, MeshLib::MeshItemType::Cell}};
+    std::array<std::vector<std::size_t>, 2> const id_map{
+        {node_ids_map, element_ids_map}};
+    std::array<std::size_t, 2> const vec_size{
+        {sfc_mesh.getNumberOfNodes(), sfc_mesh.getNumberOfElements()}};
+    std::size_t vectors_copied (0), vectors_skipped (0);
+    for (std::size_t i=0; i<type.size(); ++i)
     {
-        ERR("MeshSurfaceExtraction::createSfcMeshProperties() - Scalar array \"%s\" not found.", vec_name.c_str());
-        return false;
-    }
-    std::vector<int> const& org_vec =
-        *properties.getPropertyVector<int>( vec_name, MeshLib::MeshItemType::Cell, 1);
-    std::vector<int> sfc_prop;
-    sfc_prop.reserve(sfc_mesh.getNumberOfElements());
-    for (auto bulk_id : element_ids_map)
-    {
-        sfc_prop.push_back(org_vec[bulk_id]);
+        std::vector<std::string> const& array_names = properties.getPropertyVectorNames(type[i]);
+        for (std::string const& name : array_names)
+        {
+            if (properties.existsPropertyVector<double>(name, type[i], 1))
+            {
+                std::vector<double> const& org_vec =
+                    *properties.getPropertyVector<double>(name, type[i], 1);
+                std::vector<double> sfc_prop;
+                sfc_prop.reserve(vec_size[i]);
+                for (auto bulk_id : id_map[i])
+                {
+                    sfc_prop.push_back(org_vec[bulk_id]);
+                }
+                MeshLib::addPropertyToMesh<double>(sfc_mesh, name, type[i], 1, sfc_prop);
+                vectors_copied++;
+            }
+            else if (properties.existsPropertyVector<int>(name, type[i], 1))
```

Comment:
Handling of `int`-properties looks identical to the handling of the `double`-properties.
Reducing the code duplication by extracting the code in a function would be great.

Reply:
:heavy_check_mark: